### ISSUE TITLE
MIRAGE Inference Results

### ISSUE THREAD

**Issue by Atif-Anwer** [j-morano/MIRAGE#2]
Hi! 

Firstly, Thankyou for your fantastic work and sharing the models+code! I was able to run the scripts (`mirage_hf.py` and `mirage_wrapper.py`) successfully for inference and test on a few of the Dataset linked. However, there is ambiguity over the results being generated. 

For Eg: 
- For Segmentation, the resulting image on a test subject from the Duke_DME dataset are attached. As you can see, it is clearly extremely pixelated and contains no useful data. 
- For Classification, the terminal output for a test subject from the Duke_iAMD dataset are also attached below. I assume that the output tensors are the probabilities how MIRAGE returns a Dictionary with bscan, slo and bscanlayermap data, but there is no indication if that assumption is correct. 

So it would be great if you can clarify these issues please or provide a little more insight. I intend to finetune MIRAGE aswell on publicly available as well as a custom dataset using the provided scripts and will update if any issues arise. 

```
Creating model: pretrain_multimae_base for inputs ['bscan', 'slo', 'bscanlayermap'] and outputs ['bscan', 'slo', 'bscanlayermap']
Image size: (512, 512), Patch size: (32, 32)
Image size: (512, 512), Patch size: (32, 32)
Image size: (128, 128), Patch size: (8, 8)
Building 2D sincos posemb with h=16 and w=16
Building 2D sincos posemb with h=16 and w=16
Building 2D sincos posemb with h=16 and w=16
  Missing keys: 0
  Unexpected keys: 0
Using device: cpu
Input: Farsiu_Ophthalmology_2013_Control_Subject_1008 torch.float32 torch.Size([1, 1, 512, 512]) tensor(1.7064e-05) tensor(1.0000)
Input: bscan torch.Size([1, 1, 512, 512]) tensor(1.7064e-05) tensor(1.0000)
Input: slo torch.Size([1, 1, 512, 512]) tensor(0.) tensor(0.)
Input: bscanlayermap torch.Size([1, 128, 128]) tensor(0) tensor(0)
Outputs:
         bscan torch.Size([1, 1, 512, 512]) tensor(0.1540) tensor(0.7725)
         slo torch.Size([1, 1, 512, 512]) tensor(0.2465) tensor(0.4367)
         bscanlayermap torch.Size([1, 13, 128, 128]) tensor(-47.7551) tensor(60.1582)
```

Updated 01:
Running to run the `run_seg_eval.py` requires predictions on the test dataset. The output masks generated by `mirage_wrapper.py` file are 128x128 (for both `base` and `large` models)whereas the Ground Truths in the datasets are non-square and approx 512x512. Is the output of the model fixed to 128x128, or is there a way to upscale the predicted segmentation mask which i missed? 
Oh, and there is no evaluation script for classification also, and a rudimentary script would be really appreciated, to accommodate the multi-class classifications in atleast some of the various datasets in `classification_benchmark.md`

Thanks!

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/9373ba02-e8e4-429f-9bfd-284f62e49b56" />

**Comment by j-morano** [comment]
Hello! Thank you very much for your feedback! However, please open different issues for different topics; that way is easier for me to keep track. I added a commit to solve the first part of the issue. There was a small bug in the code of `mirage_wrapper.py`, so thank you again! I attach here the sample I tested from Duke DME, in the following order: input, OCT output, layer output.

<img width="128" height="128" alt="Image" src="https://github.com/user-attachments/assets/6ecd05d2-8f51-4e21-9dbc-85cfa721042f" />
<img width="128" height="128" alt="Image" src="https://github.com/user-attachments/assets/71a12f8b-ddb5-4c6b-80e2-3ae5c62a8c80" />
<img width="128" height="128" alt="Image" src="https://github.com/user-attachments/assets/adf3ee73-d883-4c40-83a7-b37c7cfb1c7e" />

Regarding the second part of the issue, please notice that the backbone, though it was pretrained using pseudo-labels of retinal layer segmentation, it is not supposed to be used directly for this task, as it was only employed for improving representation learning. During pretraining, the coarse pseudo-labels of retinal layers are used with 128x128 resolution for efficiency reasons.

Finally, there is no separate script for evaluating classification, because the evaluation is integrated in the `run_cls_tuning.py` script.

I hope this answers all your questions. Otherwise, do not hesitate to reopen the issue or open a new one.
